stages/org.osbuild.chown

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3
+"""
+Change file owner and group
+Change the file user and/or group ownership of one or more files or directories inside the tree.
+"""
+
+import os
+import subprocess
+import sys
+
+import osbuild.api
+from osbuild.util.path import in_tree
+
+SCHEMA = r"""
+"additionalProperties": false,
+"required": ["paths"],
+"properties": {
+  "paths": {
+    "type": "array",
+    "description": "Paths to operate on for changing ownership",
+    "items": {
+      "type": "string",
+      "pattern": "^\\/(?!\\.\\.)((?!\\/\\.\\.\\/).)+$"
+    }
+  },
+  "user": {
+    "type": "string"
+  },
+  "group": {
+    "type": "string"
+  },
+  "recursive": {
+    "type": "boolean"
+  }
+}
+"""
+
+def main(tree, options):
+    paths = options["paths"]
+    user = options.get("user", "")
+    group = options.get("group", "")
+    recursive = options.get("recursive", False)
+
+    realpaths = list()
+    for p in paths:
+        realpath = os.path.join(tree, p.lstrip("/"))
+        if not in_tree(realpath, tree, must_exist=True):
+            print(f"Error: path {p} not in tree")
+            return 1
+        realpaths.append(realpath)
+
+    arguments = ["--", f"{user}:{group}"]
+    if recursive:
+        arguments.append("--recursive")
+
+    arguments.extend(realpaths)
+
+    subprocess.run(["chown", *arguments], check=True)
```

Comment:
This stage will have to call `chown` from the image using `chroot`. An example of a stage doing the same is https://github.com/osbuild/osbuild/blob/main/stages/org.osbuild.authselect. This fact should be also noted in the doc text at the top of this file.

The reason to run the tool from the image in chroot is that it needs to be able to set user or group to the values that may exist only in the image (e.g. a custom user created by an osbuild stage).

Reply:
I'd like a `--` in between our arguments and the `realpaths`.

Reply:
Hm, somehow missed your comment about `chroot` before I made mine about parsing `passwd`.  I'm not sure what edge cases we might run into with either approach (chroot vs parsing).  Tom had some concerns about `chroot` (https://github.com/osbuild/osbuild/pull/641#discussion_r626908945) but maybe it's enough for now and we can rework it later.

Reply:
There are already stages that execute commands using `chroot` (e.g. `org.osbuild.authselect`), because the relevant tool can't work on a different FS root.

We definitely want to avoid it if we can, because this relies on the tool being available in the image itself (which is expected in most cases, but not guaranteed) and we can't ensure its presence by installing it in the build pipeline. I think that downsides are what I mentioned in the previous sentence, as well as potentially different behavior of the stage by executing the tool in potentially changing environment, which is the image FS tree.

Happy to hear @teg PoV 😉